### PR TITLE
SHS-5641: Prevent Site editors from deleting and removing imported media

### PR DESCRIPTION
**Pull request by codechefmarc**
# READY FOR REVIEW

## Summary
Removes the ability for non-administrators from deleting media that is attached to nodes that have been imported via a migration
* Removes the "Delete Media item" contextual menu item in the node edit form
* Removes the "Remove" button on the media edit form
* Only applies to non-adminstrators
* Only applies to media that was part of a migration
* This PR replaces a previous PR where the code was located on the `stanford_migrate` module. This PR only includes code changes in the `suhumsci` repo - specifically the `hs_migrate` module.

**NOTE**: Tugboat preview sites don't seem to contain any imported nodes / media. So, not exactly sure how to test other than pulling this branch locally and using the `physics` site or if there is a way to tell Tugboat to use a specific site - use `physics` to test.

## Need Review By (Date)
2024-08-28

## Urgency
Medium

## Steps to Test
1. Login to a site as a non-admin
3. Make sure the site contains migrated nodes with media entities attached. Development and testing was done with the Physics site.
4. Edit a node that has been imported via a migration. Content types used for development and testing are Person and Event. You can tell if the node was part of a migration because there are some fields that are disabled and have an orange background:

Screenshot 2024-05-30 at 3 55 31 PM

5. Click on the contextual menu for the Image or Profile Image field
6. Verify that the "Delete Media item" is no longer available

Screenshot 2024-05-30 at 3 56 37 PM

7. Click "Edit media item"
8. Verify that on the media edit form, there is new messaging at the top informing that the media can not be deleted because it was part of a migration:

Screenshot 2024-05-30 at 3 57 46 PM

9. Verify that the "Remove" button is not available next to the media item name:

Screenshot 2024-05-30 at 3 58 26 PM

10. Verify that other fields for the image can be edited, such as Focal Point, alt text, and caption.
11. Optionally, log in to the site as an admin and verify that admins can delete media from the contextual menu and from the media edit form.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207858685626376

### CODE REVIEW

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1609-xqjuxfyccjlf7llwdvhjv1s14ckv09k8.tugboatqa.com
- https://hs-colorful-xqjuxfyccjlf7llwdvhjv1s14ckv09k8.tugboatqa.com
- https://hs-traditional-xqjuxfyccjlf7llwdvhjv1s14ckv09k8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66cd0030562a46f6ec1654a5

**Review by ahughes3 — APPROVED** [review]
I was able to review imported content using the settings from the physics site, and running the importer. I masqueraded as a non developer role and the testing worked as documented.

**Review by joegl — CHANGES_REQUESTED** [review]
@codechefmarc For consistency can we get all PHP variable names to be `$snake_case` instead of `$camelCase`? Otherwise I'm ready to merge.

**Comment by codechefmarc** [comment]
@joegl - OK, changed all to snake case.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1609-xqjuxfyccjlf7llwdvhjv1s14ckv09k8.tugboatqa.com
- https://hs-colorful-xqjuxfyccjlf7llwdvhjv1s14ckv09k8.tugboatqa.com
- https://hs-traditional-xqjuxfyccjlf7llwdvhjv1s14ckv09k8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66cd0030562a46f6ec1654a5

File: docroot/modules/humsci/hs_migrate/hs_migrate.module [hunk 1]
```diff
@@ -42,3 +43,89 @@ function hs_migrate_config_pages_presave(ConfigPagesInterface $config_page) {
...
+      }
+    }
+
+  }
+}
+
+/**
+ * Checks to see if a given mediaID is associated with a migration.
+ *
+ * @var int $mediaID
```
Comment (joegl) [inline docroot/modules/humsci/hs_migrate/hs_migrate.module]: One last thing quick `mediaID` to `media_id`

```suggestion
 * Checks to see if a given media_id is associated with a migration.
 *
 * @var int $media_id
```

Comment (codechefmarc) [reply docroot/modules/humsci/hs_migrate/hs_migrate.module]: Gah, missed that one. OK, fixed. :)

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1609-xqjuxfyccjlf7llwdvhjv1s14ckv09k8.tugboatqa.com
- https://hs-colorful-xqjuxfyccjlf7llwdvhjv1s14ckv09k8.tugboatqa.com
- https://hs-traditional-xqjuxfyccjlf7llwdvhjv1s14ckv09k8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66cd0030562a46f6ec1654a5

File: docroot/modules/humsci/hs_migrate/hs_migrate.module (hunk 1 above)
Comment (joegl) [reply docroot/modules/humsci/hs_migrate/hs_migrate.module]: Thanks Marc!

**Comment by joegl** [comment]
@ahughes3 This is in!

@codechefmarc Thanks for all the work on this and being flexible in moving it back from the fork.

**Pull request merged**